datacube/api/core.py

```diff
@@ -328,11 +327,12 @@ def find_datasets(self, **search_terms):
         """
         return list(self.find_datasets_lazy(**search_terms))
 
-    def find_datasets_lazy(self, limit=None, **kwargs):
+    def find_datasets_lazy(self, ensure_location=False, limit=None, **kwargs):
```

Comment:
when  adding new keyword-based arguments it is usually safest to add them at the end (just before `**kwargs`) in case people are relying on argument order instead of names when calling this function. While no code in `datacub-core` calls this function that way, some users might. It's probably fine in this specific case though.

Reply:
fair enough